src/io/metadata.cpp

```diff
@@ -528,6 +555,32 @@ void Metadata::LoadWeights() {
   weight_load_from_file_ = true;
 }
 
+void Metadata::LoadPositions() {
+  num_positions_ = 0;
+  std::string position_filename(data_filename_);
+  // default position file name
+  position_filename.append(".position");
+  TextReader<size_t> reader(position_filename.c_str(), false);
+  reader.ReadAllLines();
+  if (reader.Lines().empty()) {
+    return;
+  }
+  Log::Info("Loading positions...");
+  num_positions_ = static_cast<data_size_t>(reader.Lines().size());
+  positions_ = std::vector<size_t>(num_positions_);
+  position_ids_ = std::vector<std::string>();
+  std::unordered_map<std::string, size_t> map_id2pos;
+  for (data_size_t i = 0; i < num_positions_; ++i) {
+    if (map_id2pos.count(reader.Lines()[i]) == 0) {
```

Comment:
What about store `reader.Lines()[i]` in a variable, since it is used multiple times?